xla/service/call_inliner.cc

```diff
@@ -176,6 +176,10 @@ bool InlineComposites(
              instruction->frontend_attributes().map().at("composite.name"));
 }
 
+bool InlineNcclGroups(HloInstruction* instruction) {
+  return !instruction->frontend_attributes().map().contains(kNcclGroupAttr);
```

Comment:
I believe this function & the corresponding test case in `call_inliner_test.cc` are unnecessary since the runtime behavior in XLA:GPU is the same for inlined vs. un-inlined functions. Both are essentially inlined for buffer assignment purpose and resulting `ThunkSequence` is the same.

Reply:
We do end up needing this, but for a different reason than you state. 

The call operation is how we define the nccl grouping boundaries. If we don't include it in the inline exceptions, then the call will disappear very early in the compiler stack and we won't have a clear way to define our NCCL group during the async wrapper pass. Ultimately, the call operation does get deleted during the async wrapper pass, so it won't survive all the way to the thunk executions.

Reply:
Got it. Thanks for the clarification!